app/database/operator/server_data_operator/handlers/thread.ts

```diff
@@ -0,0 +1,152 @@
+
+import type {HandleThreadsArgs, HandleThreadParticipantsArgs, ProcessRecordResults} from '@typings/database/database';
+import type ThreadModel from '@typings/database/models/servers/thread';
+import type ThreadParticipantModel from '@typings/database/models/servers/thread_participant';
+
+const {
+    THREAD,
+    THREAD_PARTICIPANT,
+} = Database.MM_TABLES.SERVER;
+
+export interface ThreadHandlerMix {
+    handleThreads: ({threads, prepareRecordsOnly}: HandleThreadsArgs) => Promise<Model[]>;
+    handleThreadParticipants: ({threadsParticipants, prepareRecordsOnly}: HandleThreadParticipantsArgs) => Promise<ThreadParticipantModel[]>;
+}
+
+const ThreadHandler = (superclass: any) => class extends superclass {
+    /**
+     * handleThreads: Handler responsible for the Create/Update operations occurring on the Thread table from the 'Server' schema
+     * @param {HandleThreadsArgs} handleThreads
+     * @param {Thread[]} handleThreads.threads
+     * @param {boolean | undefined} handleThreads.prepareRecordsOnly
+     * @returns {Promise<void>}
+     */
+    handleThreads = async ({threads, prepareRecordsOnly = false}: HandleThreadsArgs): Promise<Model[]> => {
+        const tableName = THREAD;
+
+        if (!threads.length) {
+            return [];
+        }
+
+        // Get unique threads in case they are duplicated
+        const uniqueThreads = getUniqueRawsBy({
+            raws: threads,
+            key: 'id',
+        }) as Thread[];
+
+        const threadsParticipants: ParticipantsPerThread[] = [];
+
+        // Let's process the thread data
+        for (const thread of uniqueThreads) {
+            threadsParticipants.push({
+                thread_id: thread.id,
+                participants: (thread.participants || []).map((participant) => ({
+                    id: participant.id,
+                    thread_id: thread.id,
+                })),
+            });
+        }
+
+        // Process the threads to get which ones need to be created and which updated
+        const processedThreads = (await this.processRecords({
+            createOrUpdateRawValues: uniqueThreads,
+
+            // deleteRawValues: pendingThreadsToDelete,
+            tableName,
+            findMatchingRecordBy: isRecordThreadEqualToRaw,
+            fieldName: 'id',
+        })) as ProcessRecordResults;
+
+        const preparedThreads = (await this.prepareRecords({
+            createRaws: processedThreads.createRaws,
+            updateRaws: processedThreads.updateRaws,
+            deleteRaws: processedThreads.deleteRaws,
+            transformer: transformThreadRecord,
+            tableName,
+        })) as ThreadModel[];
+
+        // Add the models to be batched here
+        const batch: Model[] = [...preparedThreads];
+
+        // calls handler for Thread Participants
+        const threadParticipants = (await this.handleThreadParticipants({threadsParticipants, prepareRecordsOnly: true})) as ThreadParticipantModel[];
```

Comment:
handleThreadParticipants can throw, so only call it if you have threadsParticipants

Reply:
`threadsParticipants` will at least one item as it is created per thread even in case there are no participants.

Reply:
Is it because of the post author?

Reply:
`threadsParticipants` type is
```
type ParticipantsPerThread = {
    thread_id: string;
    participants: ThreadParticipant[];
};
```

As this function will not come until here without a single thread, we will at least have a single item for the thread.